src/config.rs

```diff
@@ -1494,45 +1493,70 @@ pub enum DBType {
     Raft,
 }
 
-pub struct DBConfigManger {
-    db: RocksEngine,
+pub struct DBConfigManger<T: TabletFactory<RocksEngine>> {
+    tablet_factory: Arc<T>,
     db_type: DBType,
     shared_block_cache: bool,
 }
 
-impl DBConfigManger {
-    pub fn new(db: RocksEngine, db_type: DBType, shared_block_cache: bool) -> Self {
+impl<T: TabletFactory<RocksEngine>> DBConfigManger<T> {
+    pub fn new(tablet_factory: Arc<T>, db_type: DBType, shared_block_cache: bool) -> Self {
         DBConfigManger {
-            db,
+            tablet_factory,
             db_type,
             shared_block_cache,
         }
     }
-}
 
-impl DBConfigManger {
     fn set_db_config(&self, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
-        self.db.set_db_options(opts)?;
-        Ok(())
+        let mut result = Ok(());
+        self.tablet_factory.for_each_opened_tablet(Box::new(
+            |region_id, suffix, db: &RocksEngine| {
+                let r = db.set_db_options(opts);
+                if r.is_err() {
+                    result = Err(Box::from(r.err().unwrap()));
+                    error!(
+                        "set_db_config failed on tablet {}_{}. err:{:?}",
+                        region_id, suffix, &result
+                    );
+                }
+            },
+        ));
+        result
     }
 
     fn set_cf_config(&self, cf: &str, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
+        let mut result = Ok(());
         self.validate_cf(cf)?;
-        self.db.set_options_cf(cf, opts)?;
-        // Write config to metric
-        for (cfg_name, cfg_value) in opts {
-            let cfg_value = match cfg_value {
-                v if *v == "true" => Ok(1f64),
-                v if *v == "false" => Ok(0f64),
-                v => v.parse::<f64>(),
-            };
-            if let Ok(v) = cfg_value {
-                CONFIG_ROCKSDB_GAUGE
-                    .with_label_values(&[cf, cfg_name])
-                    .set(v);
+        self.tablet_factory.for_each_opened_tablet(Box::new(
+            |region_id, suffix, db: &RocksEngine| {
+                let r = db.set_options_cf(cf, opts);
+                if r.is_err() {
+                    result = Err(Box::from(r.err().unwrap()));
+                    error!(
```

Comment:
If it fails on all db, then it will write region count lines of logs? Doesn't seem to be reasonable.

Reply:
These APIs call should never fail in legitimate scenarios.  I would rather see the the error! itself have some mechanism to ignore the error messages instead of doing some limit in the caller side everywhere.

Reply:
> the error! itself have some mechanism to ignore the error messages 

How?

Reply:
Basically we need to know all the failed tablet/cfs that do not apply the new config.

Reply:
Do we really need to know that? Isn't know the failure count and some of errors sufficient?